### ISSUE TITLE
inputs.artifacts[].git.depth not work

### ISSUE THREAD

**Issue by hidekuro** [argoproj/argo#1524]
**Is this a BUG REPORT or FEATURE REQUEST?**:

BUG REPORT

**What happened**:

here is workflow example.

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: issue-sample-
spec:
  entrypoint: git-clone
  templates:
    - name: git-clone
      inputs:
        artifacts:
          - name: argo-source
            path: /workspace
            git:
              repo: https://github.com/argoproj/argo.git
              revision: master
              depth: 1
      container:
        image: buildpack-deps:bionic-scm
        command: ["sh", "-c"]
        args: ["git status"]
        workingDir: /workspace
```

and lint it by `argo lint issue-sample.yaml`, I got error.

```
FATA[0000] issue-sample.yaml failed to parse: error unmarshaling JSON: while decoding JSON: json: unknown field "depth"
```

It seems `depth: 1` doesn't work.

**What you expected to happen**:

No errors.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Argo version:
```
$ argo version

argo: v2.3.0
  BuildDate: 2019-05-20T22:11:23Z
  GitCommit: 88fcc70dcf6e60697e6716edc7464a403c49b27e
  GitTreeState: clean
  GitTag: v2.3.0
  GoVersion: go1.11.5
  Compiler: gc
  Platform: darwin/amd64
```
- Kubernetes version :
```
$ kubectl version -o yaml

clientVersion:
  buildDate: "2019-07-18T14:25:20Z"
  compiler: gc
  gitCommit: 4485c6f18cee9a5d3c3b4e523bd27972b1b53892
  gitTreeState: clean
  gitVersion: v1.15.1
  goVersion: go1.12.7
  major: "1"
  minor: "15"
  platform: darwin/amd64
serverVersion:
  buildDate: "2019-06-19T16:32:14Z"
  compiler: gc
  gitCommit: e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529
  gitTreeState: clean
  gitVersion: v1.15.0
  goVersion: go1.12.5
  major: "1"
  minor: "15"
  platform: linux/amd64
```

**Comment by hidekuro** [comment]
I tried `kubctl apply -f issue-sample.yaml`. it succeed.
therefore, this is lint issue maybe.

**Comment by jessesuen** [comment]
The depth feature was just merged and will be in v2.4 (unreleased). If you try `:latest` of the controller and executor, it should work.

**Comment by hidekuro** [comment]
thx for quick response. resolved it.
